src/Orchard.Cms.Web/Modules/Orchard.Settings/Controllers/AdminController.cs

```diff
@@ -67,11 +75,34 @@ public class AdminController : Controller, IUpdateModel
                 await _siteService.UpdateSiteSettingsAsync(siteSettings);
 
                 _notifier.Success(H["Site settings updated successfully."]);
-                
+
                 return RedirectToAction(nameof(Index), new { groupId });
             }
 
             return View(viewModel);
         }
+
+        public async Task<IActionResult> RestartSite()
+        {
+            if (!await _authorizationService.AuthorizeAsync(User, Permissions.Restart))
+            {
+                return Unauthorized();
+            }
+
+            return View();
+        }
+
+        [HttpPost]
+        [ActionName(nameof(RestartSite))]
+        public async Task<IActionResult> RestartSitePost()
+        {
+            if (!await _authorizationService.AuthorizeAsync(User, Permissions.ManageSettings))
+            {
+                return Unauthorized();
+            }
+            _orchardHost.ReloadShellContext(_shellSettings);
+
+            return Redirect("~/admin");
```

Comment:
Don't use absolute urls if it's not the homepage, prefer an action name. In this case, just return on the same page `RedirectToAction("RestartSite")`